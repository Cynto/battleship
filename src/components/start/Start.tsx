import React from 'react';
import './start.css';
import { useSelector, useDispatch } from 'react-redux';

function Start() {
  const playerGridArray = useSelector((state: any) => state.playerGridArray);
  console.log(playerGridArray);
  return (
    <div data-testid="start" className="start">
      <div className="grid-layout-title-container">
        <div className="grid-layout-container">
          <div className="grid-container"></div>
        </div>
      </div>
    </div>
  );
}

export default Start;
