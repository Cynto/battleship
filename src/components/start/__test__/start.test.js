import React from 'react';
import ReactDOM from 'react-dom';
import Start from '../Start';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { HashRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';

describe('Main Container tests', () => {
  afterEach(cleanup);
  test('Renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <Provider>
        <Router>
          <Start />{' '}
        </Router>
      </Provider>,
      div,
    );
  });
  test('Renders main container correctly', () => {
    const { getByTestId } = render(
      <Provider>
        <Router>
          <Start />{' '}
        </Router>
      </Provider>,
    );
    expect(getByTestId('start')).toHaveClass('start');
  });
  test('Matches snapshot', () => {
    const tree = renderer
      .create(
        <Router>
          <Start />
        </Router>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
