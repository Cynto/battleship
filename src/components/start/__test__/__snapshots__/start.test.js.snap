// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main Container tests Matches snapshot 1`] = `
<div
  className="start"
  data-testid="start"
>
  <div
    className="grid-layout-container"
  >
    <div
      className="grid-container"
    />
  </div>
</div>
`;
